auth/basic/db.go

```diff
@@ -60,6 +63,42 @@ func (b *Store) CreateTx(ctx context.Context, tx *sql.Tx, userID, username, pass
 	return err
 }
 
+// UpdateTx should update auth_basic_users with newPassword if username and oldPassword matches
+func (b *Store) UpdateTx(ctx context.Context, tx *sql.Tx, userID, username, oldPassword, newPassword string) error {
+	err := permission.LimitCheckAny(ctx, permission.System, permission.Admin, permission.User, permission.MatchUser(userID))
+	if err != nil {
+		return err
+	}
+
+	err = validate.Many(
+		validate.UUID("UserID", userID),
+		validate.Username("Username", username),
+		validate.Text("oldPassword", oldPassword, 0, 200),
+		validate.Text("newPassword", newPassword, 8, 200),
+	)
+	if err != nil {
+		return err
+	}
+
+	if !permission.Admin(ctx) {
+		validatedUserId, err := b.Validate(ctx, username, oldPassword)
+		if err != nil {
+			return validation.NewFieldError("oldPassword", "Invalid Password")
+		}
+
+		if validatedUserId != userID {
+			return err
+		}
+	}
+
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(newPassword), 12)
```

Comment:
This would create a new occurrence of bug #2985 

We should merge #2990 first which fixes it, and then update this PR to use the `HashedPassword` type.